src/components/Conversion/ConversionNOB.vue

```diff
@@ -0,0 +1,143 @@
+                @click="onSubmitClicked()"
+              >
+                <span>Done</span>
+              </v-btn>
+              <v-btn large outlined color="primary" id="nob-cancel-btn"
+                @click="onCancelClicked()"
+              >
+                <span>Cancel</span>
+              </v-btn>
+            </div>
+          </v-form>
+        </div>
+
+        <div v-if="!onEditMode" class="summary-block d-flex justify-space-between align-center">
+          <span id="naics-summary">{{ naicsSummary }}</span>
+          <v-btn text color="primary" id="nob-change-btn" @click="onChangeClicked()">
+            <v-icon small>mdi-pencil</v-icon>
+            <span>Change</span>
+          </v-btn>
+        </div>
+      </v-col>
+    </v-row>
+  </section>
+</template>
+
+<script lang="ts">
+import { Action, Getter } from 'vuex-class'
+import { Component, Vue } from 'vue-property-decorator'
+import { ActionBindingIF } from '@/interfaces/'
+import { NaicsIF } from '@bcrs-shared-components/interfaces/'
+import { isEqual } from 'lodash'
+
+@Component({})
+export default class NatureOfBusiness extends Vue {
+  @Getter getCurrentNaics!: NaicsIF
+  @Getter hasNatureOfBusinessChanged!: boolean
+
+  @Action setNaics!: ActionBindingIF
+
+  // local variables
+  private onEditMode = false
+  private maxLength = 300
+  private naicsText = ''
+  private naicsRules = [
+    (v: string) => (v?.length <= this.maxLength) || 'Maximum 300 characters reached'
+  ]
+
+  /** Show naics value, description or (Not Entered) upon first render */
+  get naicsSummary (): string {
+    const code = this.getCurrentNaics.naicsCode
+    const desc = this.getCurrentNaics.naicsDescription
+    let summary = '(Not Entered)'
+    if (code && desc) {
+      this.naicsText = this.hasNatureOfBusinessChanged ? this.naicsText : `${code} - ${desc}`
+      summary = `${code} - ${desc}`
+    } else if (desc) {
+      this.naicsText = desc
+      summary = desc
+    }
+    return summary
+  }
+
+  /** The naics data on record for the business. */
+  get originalNaics (): NaicsIF {
+    return {
+      naicsCode: this.getCurrentNaics.naicsCode,
+      naicsDescription: this.getCurrentNaics.naicsDescription
+    }
+  }
+
+  /** Called when user has clicked the Change button. */
+  onChangeClicked (): void {
+    this.onEditMode = true
+  }
+
+  /** Submited when user has clicked the Done button. */
+  onSubmitClicked (): void {
+    let validForm = (this.$refs.form as Vue & { validate: () => boolean }).validate()
+    if (validForm) {
+      if (!isEqual(this.naicsText, this.naicsSummary)) {
+        this.setNaics({
+          naicsCode: '',
+          naicsDescription: this.naicsText
+        })
+      }
+      this.onEditMode = false
+    }
+  }
+
+  /** Called when user has clicked the Cancel button. */
+  onCancelClicked (): void {
+    this.setNaics(this.originalNaics)
+    this.onEditMode = false
+  }
+}
+</script>
```

Comment:
Looks really good now.

I think you need to emit an event to inform the parent whether this form is valid.

Reply:
^^ This component is invalid if the required data has not been entered, of if the form is still open (edit mode). You can look at other components for examples.

Reply:
This form is not required, but I will do an emit to check if the form is still open.

Reply:
I'm not sure what to display since there is no UX/UI for it. If we click on "Review and Confirm" when the form is still open, it will be closed automatically.

Reply:
However, I missed the "undo" button, and I will add that now.

Reply:
It's a general rule that a component is not valid if it is still "open".

Is it possible that the user (staff) enters an empty NAICS description? Would that be invalid?

Reply:
I added validation for edit mode and more in my latest commit. The staff can leave the text area empty, and it will not raise an error. If the text area is left empty, the summary will display "(Not Entered)".